galaxy/main/migrations/0126_community_score_question_average.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.17 on 2018-12-10 19:33
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+SET_COMMUNITY_SCORE_TO_QUESTION_AVERAGE = """
+UPDATE main_repository
+set community_score =
+    (SELECT (((sum(docs-1))/4.0 +
+             sum(ease_of_use-1)/4.0 +
+             sum(does_what_it_says-1)/4.0 +
+             sum(works_as_is-1)/4.0 +
+             sum(used_in_production-1)/4.0
+             ) / (count(docs) + count(ease_of_use) + count(does_what_it_says) +
+                  count(works_as_is) + count(used_in_production)))
+            * 5 as average
+    FROM main_communitysurvey
+        WHERE main_communitysurvey.repository_id = main_repository.id)
+"""
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('main', '0125_collection_base'),
+    ]
+
+    operations = [
+        migrations.RunSQL(SET_COMMUNITY_SCORE_TO_QUESTION_AVERAGE),
+    ]
```

Comment:
If `update_community_score()` where part of the Model or otherwise easily available outside the view itself, we could update the existing repository scores using the same code path you've already updated in this PR, ensuring no chance of a mismatch between two different implementations of the algorithm in different languages, one of which exists only in a migration and will only be run once. (Notoriously difficult to validate!)

This would also be a safer setup for future changes to the scoring algorithm, which has already changed a few times and surely will continue to do so.

Reply:
Thoughts:
1. Performing data migrations on can be bottleneck and affect service availability.
2. Running python will affect performance drastically, because instead of single SQL we will end up with sending data back and forth within the single transaction.
3. Re-using code in migration is dangerous, because code can eventually change that WILL break the migration.
4. The best option in this particular case I think to create a background task to recalculate community scores and run it after migration.